marketplaceItems/questionApp/appResources/appData/question_app.js

```diff
@@ -0,0 +1,238 @@
+    avatar's head and creates a question mark entity there */
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD_TOP = 0.3;
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD = 0.4;
+    var QUESTION_MARK_Y_OFFSET_RATIO_FROM_HIPS = 0.6;
+    var GROWTH_INTERVAL_MS = 1000; // takes 4 min 38 sec to reach height of 0.8 from 0.2 at ratio:1.005, interval: 1000
+    var MAX_HEIGHT_M = 0.8;
+    var GROWTH_RATIO = 1.1;
+    var questionMark;
+    var questionMarkMaterial;
+    var growingInterval;
+    function createQuestionMark() {
+        if (questionMark) {
+            return;
+        }
+        var avatarHeight = MyAvatar.getHeight();
+        var parentJointIndex = MyAvatar.getJointIndex("HeadTop_End");
+        var offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD_TOP;
+        print("USING HEAD TOP JOINT");
+        if (parentJointIndex === -1) {
+            print("NOPE>>>USING HEAD JOINT");
+            parentJointIndex = MyAvatar.getJointIndex("Head");
+            offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HEAD;
+        }
+        if (parentJointIndex === -1) {
+            parentJointIndex = MyAvatar.getJointIndex("Hips");
+            offsetRatio = QUESTION_MARK_Y_OFFSET_RATIO_FROM_HIPS;
+            print("ERROR: Falling back to hips joint as head could not be found");
+        }
+        var questionMarkLocalYPosition = avatarHeight * offsetRatio;
+        questionMark = Entities.addEntity({
+            name: "Question App Mark",
+            type: "Model",
+            modelURL: Script.resolvePath("resources/models/sphere-white-emissive.fbx"),
+            lifetime: 360000,
+            parentID: MyAvatar.sessionUUID,
+            parentJointIndex: parentJointIndex,
+            localPosition: { x: 0, y: questionMarkLocalYPosition, z: 0 },
+            dimensions: { x: 0.2, y: 0.2, z: 0.2 },
+            grab: { grabbable: false },
+            collisionless: true
+        }, 'avatar');
+        questionMarkMaterial = Entities.addEntity({
+            type: "Material",
+            name: "Question App Material",
+            materialURL: "materialData",
+            priority: 1,
+            parentID: questionMark,
+            materialData: JSON.stringify({
+                materials: {
+                    albedo: { red: 0, green: 255, blue: 0 },
+                    emissive: { red: 0, green: 1, blue: 0 }
+                }
+            })
+        }, 'avatar');
+        print("start growing");
+        growingInterval = Script.setInterval(function() {
+            var questionMarkDimensions = Entities.getEntityProperties(questionMark, 'dimensions').dimensions;
+            questionMarkDimensions = Vec3.multiply(questionMarkDimensions, GROWTH_RATIO);
+            Entities.editEntity(questionMark, { dimensions: questionMarkDimensions });
+            if (questionMarkDimensions.y > MAX_HEIGHT_M) {
+                print("stop growing");
+                Script.clearInterval(growingInterval);
```

Comment:
please check to make sure `growingInterval` is truthy here before clearing the interval.

also, please set `growingInterval` to falsey at this point.

Reply:
this has to be true because it is coming from inside the interval, right? Do I still need to check that? I did add in a line to set it to null tho!

Reply:
i don't THINK that is the case - i believe, if `cleanUp` is called (which would clear the interval), and the interval timer has already started, this function will get called.

even if that can't happen in this particular instance, it's still a good practice to check if a timer is still present before clearing it.